### PR TITLE
feat: add support for Deneb consensus types and fix beacon sync

### PR DESCRIPTION
**Pull request by ogenev**
### What was wrong?

1. We had a recently introduced bug in beacon sync which prevented a node from syncing with the tip of the chain.
2. To fully support post-merge Era files we need to support the Deneb fork changes introduced in consensus types.

### How was it fixed?
1. Fix the beacon sync bug
2. Update beacon block, execution payload header, and beacon state to support Deneb changes.
3. Update light client types to Deneb.

### To-Do

[//]: # (Stay ahead of things, add list items here!)
- [ ] Clean up commit history and use [conventional commits](https://www.conventionalcommits.org/en/v1.0.0/).

### CODE REVIEW

File: ethportal-api/src/types/consensus/kzg_commitment.rs [hunk 1]
```diff
@@ -0,0 +1,138 @@
...
+        Self::from_str(&string).map_err(serde::de::Error::custom)
+    }
+}
+
+impl FromStr for KzgCommitment {
+    type Err = String;
+
+    fn from_str(s: &str) -> Result<Self, Self::Err> {
+        if let Some(stripped) = s.strip_prefix("0x") {
+            let bytes = hex::decode(stripped).map_err(|e| e.to_string())?;
```
Comment (njgheorghita) [inline ethportal-api/src/types/consensus/kzg_commitment.rs:95]: why not use `ethportal_api::utils::bytes::hex_decode`?

File: ethportal-api/src/types/consensus/kzg_commitment.rs [hunk 2]
```diff
@@ -0,0 +1,138 @@
...
+            }
+        } else {
+            Err("must start with 0x".to_string())
+        }
+    }
+}
+
+impl Debug for KzgCommitment {
+    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+        write!(f, "{}", serde_utils::hex::encode(self.0))
```
Comment (njgheorghita) [inline ethportal-api/src/types/consensus/kzg_commitment.rs:115]: same with `ethportal_api::utils::bytes::hex_encode()`

File: ethportal-api/src/types/consensus/kzg_commitment.rs [hunk 3]
```diff
@@ -0,0 +1,138 @@
...
+    }
+}
+
+#[test]
```
Comment (njgheorghita) [inline ethportal-api/src/types/consensus/kzg_commitment.rs:119]: I think we should continue the convention of using `mod tests {}` as a wrapper around the tests, which is how we typically do it across the code base.

File: ethportal-api/src/types/consensus/kzg_commitment.rs [hunk 4]
```diff
@@ -0,0 +1,138 @@
...
+        }
+        write!(f, "…")?;
+        for i in &self.0[BYTES_PER_COMMITMENT - 2..BYTES_PER_COMMITMENT] {
+            write!(f, "{i:02x}")?;
+        }
+        Ok(())
+    }
+}
+
+impl TreeHash for KzgCommitment {
```
Comment (njgheorghita) [inline ethportal-api/src/types/consensus/kzg_commitment.rs:53]: It seems worth adding tests specifically for `tree_hash_root()` if we're custom implementing this trait

File: light-client/src/consensus/consensus_client.rs [hunk 5]
```diff
@@ -384,6 +389,9 @@ impl<R: ConsensusRpc> ConsensusLightClient<R> {
 
         let should_apply_update = {
             let has_majority = committee_bits * 3 >= 512 * 2;
+            if !has_majority {
```
Comment (njgheorghita) [inline light-client/src/consensus/consensus_client.rs:392]: is this something you meant to leave in?

**Review by njgheorghita — APPROVED** [review]

**Review by KolbyML — APPROVED** [review]
:shipit:

File: light-client/src/consensus/consensus_client.rs (hunk 5 above)
Comment (ogenev) [reply light-client/src/consensus/consensus_client.rs:392]: yes, updated the log message to be more clear.

**Pull request merged**